### ISSUE TITLE
Workflows should be schedulable like cronjob

### ISSUE THREAD

**Issue by sdminonne** [sdminonne/workflow-controller#29]
Long story short. One should be able to schedule a workflow like kube cronjob.

**Comment by sdminonne** [comment]
Assigning to me since I'm writing the proposal for this.

**Comment by davidc-donorschoose** [comment]
I see that a "cron workflow controller: new code" commit went in last month (on December 7, 2018). I don't see any examples in the commit.

We are definitely interested in using this feature. We did a quick merge from your repo into our fork just to try it out. We can build and install the upgraded workflow-controller and even create a CronWorkflow resource (I determined the correct yaml by reading your go source code). The job doesn't appear to run as scheduled - which is completely understandable since you haven't closed this case yet.

We could probably test something for you on AWS EKS if you want.

**Comment by sdminonne** [comment]
We need to create examples in the doc for CronWorkflow and DaemonSetJob

**Comment by davidc-donorschoose** [comment]
Apologies for the delay. I realized that we tested with a manifest that left out a couple nested yaml tags under the "workflowTemplate". We'll fix this and re-test, which I'm pretty sure will work. Then I'll get you a gist with our yaml that you could consider as a example.

**Comment by davidc-donorschoose** [comment]
My CronWorkflow didn't run as expected. I created it with the manifest in this gist (https://gist.github.com/davidc-donorschoose/79bb22c203000d4ecea702618923e709). I used UTC for the Schedule set 5 minutes in the future when I ran the `kubectl create --namespace=qa -f /tmp/cwfl.yaml` command into my EKS cluster. I can see that it created with `kubectl describe cwfl/qa-nightly-batch --namespace=qa`. 

However, it didn't run. I don't see any Workflow, Jobs, or Pods running. 

BTW, to see Workflow resources, I have to use the fully scoped custom resource type `kubectl get workflow.workflow.k8s.io --namespace=qa`, where as I can see my CronWorkflow with the fully scoped type `cronworkflow.workflow.k8s.io`, but also with `cronworkflows`, `cronworkflow` and `cwfl`.

**Comment by sdminonne** [comment]
@davidc-donorschoose thank for the use case. Going to have a look and if you agree adding something very similar to our e2e test.

**Comment by davidc-donorschoose** [comment]
Of course, you may use the CronWorkflow manifest in that gist for any purpose. Please let me know if you find something wrong in my yaml code or if you do  get something similar to work. 

Also, please reach out if you think I can help with AWS EKS testing in some way. I suppose it is possible that EKS has some custom resource incompatibility with your CronWorkflow implementation (although we can create an ordinary Workflow and that does run).

**Comment by sdminonne** [comment]
@davidc-donorschoose thanks.
All the code we've in this repo is tested with 1.9 (if I'm not wrong) but it should work with 1.7.
It uses very basics CRDs features. In the next weeks (hopefully not months) we're going to add more features that will request more advanced CRDs features (OpenAPI validation, subresources and webhooks). Obviously we'll branch preserving 1.9 version in case someone wants to keep some old cluster.

**Comment by davidc-donorschoose** [comment]
EKS is currently on K8s-v1.10.11

**Comment by clamoriniere1A** [comment]
Hi @davidc-donorschoose 
#89 explains how to fix your CronWorkflow definition.
Regards,
